core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -249,11 +249,15 @@ class GroupCoordinator(val brokerId: Int,
             clientId, clientHost, protocolType, protocols, group, responseCallback)
         }
       } else {
-        val isKnownGroupInstanceId = group.hasStaticMember(groupInstanceId)
-        val groupInstanceIdNotFound = groupInstanceId.isDefined && !isKnownGroupInstanceId
+        val staticMemberId = if (group.hasStaticMember(groupInstanceId))
```

Comment:
This might be clearer if we separate the static and dynamic cases explicitly:
```scala
groupInstanceId match {
  case Some(instanceId) =>  
    // Static member
  case None => 
    // Dynamic member
}
```